plutus-core/untyped-plutus-core/Language/UntypedPlutusCore/Evaluation/Machine/Cek.hs

```diff
@@ -264,7 +270,7 @@ instance (Eq fun, Hashable fun, ToExMemory term) =>
             SpendBudget (CekCarryingM term uni fun) fun (ExBudgetCategory fun) term where
     spendBudget key budget = do
         modifying exBudgetStateTally
-                (<> (ExTally (singleton key budget)))
+                (<> ExTally (singleton key (Vector.singleton budget)))
```

Comment:
I wonder how this will affect the overall cost of execution. We should check.

Reply:
That _should_ be for counting mode only, where performance doesn't really matter much.